### PR TITLE
[BEAM-6767] SamzaRunner: Add more parameters to SamzaPipelineOptions

### PR DESCRIPTION
**Pull request by xinyuiscool**
Add the following parameter support:

1. Allow enable/disable beam metrics in SamzaPipelineOptions
2. Enable durable state with Samza pipeline options
3. Enable host-affinity for durable state

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by xinyuiscool** [comment]
R: @lhaiesp

File: runners/samza/src/main/java/org/apache/beam/runners/samza/translation/ConfigBuilder.java [hunk 1]
```diff
@@ -66,16 +65,26 @@ public void putAll(Map<String, String> properties) {
...
       config.putAll(createUserConfig(options));
 
       config.put(JobConfig.JOB_NAME(), options.getJobName());
+      config.put(JobConfig.JOB_ID(), options.getJobInstance());
+
       config.put(
           "beamPipelineOptions",
           Base64Serializer.serializeUnchecked(new SerializablePipelineOptions(options)));
 
+      // TODO: remove after SAMZA-1531 is resolved
```
Comment (lhaiesp) [inline runners/samza/src/main/java/org/apache/beam/runners/samza/translation/ConfigBuilder.java]: what's the reason to move it from localRunConfig to here?

File: runners/samza/src/main/java/org/apache/beam/runners/samza/translation/ConfigBuilder.java [hunk 2]
```diff
@@ -90,25 +99,16 @@ private static boolean isEmptyUserConfig(Map<String, String> config) {
...
-      throws Exception {
+  private static Map<String, String> createUserConfig(SamzaPipelineOptions options) {
     final String configFilePath = options.getConfigFilePath();
     final Map<String, String> config = new HashMap<>();
 
     // If user provides a config file, use it as base configs.
     if (StringUtils.isNoneEmpty(configFilePath)) {
       final File configFile = new File(configFilePath);
+      checkArgument(configFile.exists(), "Config file %s does not exist", configFilePath);
+      final PropertiesConfigFactory configFactory = new PropertiesConfigFactory();
```
Comment (lhaiesp) [inline runners/samza/src/main/java/org/apache/beam/runners/samza/translation/ConfigBuilder.java]: why forcing to use PropertiesConfigFactory here?

**Comment by xinyuiscool** [comment]
Run Java PreCommit

File: runners/samza/src/main/java/org/apache/beam/runners/samza/translation/ConfigBuilder.java (hunk 2 above)
Comment (xinyuiscool) [reply runners/samza/src/main/java/org/apache/beam/runners/samza/translation/ConfigBuilder.java]: Seems something got wrong in the merge. Let me rebase again with the last version.

**Comment by xinyuiscool** [comment]
Run Java PreCommit

**Comment by xinyuiscool** [comment]
Run Spotless PreCommit

**Comment by xinyuiscool** [comment]
Run Python PreCommit

**Comment by xinyuiscool** [comment]
Run Portable_Python PreCommit

**Pull request merged**